### PR TITLE
deprecated Twig_Template::getEnvironment()

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Comment by stof** [comment]
what about throwing a silenced deprecation, according to the Symfony way to handle them ?

as it is silenced, it will not hurt other projects not following this pattern (they just won't display it), unless they have a crappy error handler turning silenced errors into exceptions (which would break tons of libraries anyway)

File: doc/deprecated.rst [hunk 1]
```diff
@@ -130,3 +130,7 @@ Miscellaneous
 
 * As of Twig 1.x, ``Twig_Environment::clearTemplateCache()`` is deprecated and
   will be removed in 2.0.
+
+* As of Twig 1.x, ``Twig_Template::getEnvironment()`` and
+  ``Twig_TemplateInterface::getEnvironment()`` are deprecated and will be
+  removed in 2.0.
```
Comment (sstok) [inline doc/deprecated.rst]: The deprecation message says it's removed in 3.0.
`@deprecated since 1.20 (to be removed in 3.0)`

**Comment by stof** [comment]
@fabpot any feedback on the warning ?

**Comment by fabpot** [comment]
@stof I've just added the `trigger_error` call.

**Comment by stof** [comment]
:+1: (we should make a pass on other deprecated methods in Twig btw)

**Pull request merged**